pkg/pillar/scripts/generate-device.sh

```diff
@@ -27,7 +27,10 @@ done
 lifetime=$((365 * 20))
 dir=$(dirname "$0")
 if [ "$use_tpm" = true ]; then
-"$dir"/generate-self-signed.sh -t -b "$output_base" -l "$lifetime"
+    if ! "$dir"/generate-self-signed.sh -t -b "$output_base" -l "$lifetime"; then
+        exit 1
+    fi
 else
-"$dir"/generate-self-signed.sh -b "$output_base" -l "$lifetime"
+    "$dir"/generate-self-signed.sh -b "$output_base" -l "$lifetime"
```

Comment:
Shouldn't you check for error return in both calls?

Reply:
Or document in generate-device and generate-selfsigned that only with TPM (-t) can it possibly fail.

Reply:
Added a return value check for both the calls.